src/jit/codegenxarch.cpp

```diff
@@ -1557,7 +1557,7 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
 
             GenTreePtr op1 = treeNode->gtGetOp1();
             genConsumeRegs(op1);
-            emit->emitInsBinary(ins_Store(targetType), emitTypeSize(treeNode), treeNode, op1);
```

Comment:
@sandreenko - I agree with @pgavlin ; I hadn't noticed the addition of these two changes. This is only correct to do in the case where we can guarantee that this is a single field of the struct. This code is for the general case of `GT_STORE_LCL_FLD` for which we do not have that guarantee. When we have an explicit store of a byte in the code stream, `Lowering` should have requested a byte register.